aten/src/ATen/BatchingUtils.h

```diff
@@ -0,0 +1,42 @@
+#include <ATen/BatchedTensorImpl.h>
+
+namespace at {
+
+/*
+ * Utility functions used to implement batching rules.
+ */ 
+
+// Creates a bitset of all the levels present in `bdims`.
+//
+// For example:
+//   createLevelsBitset([(lvl=1, dim=2), (lvl=3, dim=1)]) -> 1010000...
+std::bitset<kVmapNumLevels> createLevelsBitset(BatchDimsRef bdims);
+
+// Produces a new vector of the same size as `arr` where each element of is
+// created by calling `func` on  each element of `arr` in order.
+// Equivalent to `output = map(func, arr)` in Python.
+std::vector<int64_t> transformIntVector(IntArrayRef arr, std::function<int64_t(int64_t)> func);
+
+// Takes a BatchedTensor or a regular Tensor and permutes all of the batch dims of
```

Comment:
The "or a regular Tensor" makes it sound like there is value in this function if it's not in fact a BatchedTensor. I'd write out the case for BatchedTensor and then also say that this acts as an identity operation (for convenience) if the given Tensor is not a BatchedTensor.